pbsmetrics/metrics.go

```diff
@@ -275,4 +275,5 @@ type MetricsEngine interface {
 	RecordStoredImpCacheResult(cacheResult CacheResult, inc int)
 	RecordPrebidCacheRequestTime(success bool, length time.Duration)
 	RecordRequestQueueTime(success bool, requestType RequestType, length time.Duration)
+	RecordTimeoutNotice(sucess bool)
```

Comment:
Should this be a per-adapter metric?

Reply:
Eventually it may need to be. At this point I did not want to pollute the prometheus metric space with 59 unused adapter metrics.

Reply:
Shouldn't be a breaking change. If we add an adapter label later, any Prometheus queries written against the current labels should still work, just sum over all adapters as expected. Could be missleading if we hardcode support only for facebook, and someone adds timeout support for another adapter without catching the hardcoded part in the code.

Reply:
The sum over all adapters isn't automatic though. The PromQL query would return a series of metrics instead of a single value. Perhaps we can add a new per-adapter metric in the future when needed.